src/styles/main.scss

```diff
@@ -0,0 +1,35 @@
+.main {
+  justify-content: center;
+  margin: 0;
+  padding: 50px 40px;
+  display: grid;
+
+  --column-count: 4;
```

Comment:
The CSS variable '--column-count' is correctly used to manage the number of columns in the grid layout, which aligns with the task requirement for responsive design.